File: blind-run-backend/src/main/java/com/example/demo/entity/UserRole.java
package com.example.demo.entity;

/**
 * 用户角色枚举 —— 区分盲人和志愿者身份
 */
public enum UserRole {
    /** 未设定角色（新用户默认状态） */
    UNSET,
    /** 盲人用户：可以下单预约陪跑服务 */
    BLIND,
    /** 志愿者：可以接单提供陪跑服务 */
    VOLUNTEER
}

File: blind-run-frontend/lib/core/models/user_role.dart
enum UserRole {
  unset,
  blind,
  volunteer,
}

extension UserRoleX on UserRole {
  String get storageValue => backendValue;

  String get backendValue => switch (this) {
        UserRole.unset => 'UNSET',
        UserRole.blind => 'BLIND',
        UserRole.volunteer => 'VOLUNTEER',
      };

  String get label => switch (this) {
        UserRole.unset => '待选择身份',
        UserRole.blind => '盲人跑者',
        UserRole.volunteer => '志愿者',
      };

  static UserRole? fromStorage(String? value) {
    return fromBackend(value);
  }

  static UserRole? fromBackend(String? value) {
    return switch (value?.toUpperCase()) {
      'UNSET' => UserRole.unset,
      'BLIND' => UserRole.blind,
      'VOLUNTEER' => UserRole.volunteer,
      _ => null,
    };
  }
}
